### ISSUE TITLE
truthiness when comparing with blank and empty are different when undefined

### ISSUE THREAD

**Issue by mikebridge** [Shopify/liquid#642]
I'm not sure I understand what's supposed to happen when comparing an undefined value with `empty` or with `blank`.  I thought both should evaluate to false, but this doesn't seem to be the case.  Is the following expected behaviour?

After I include activesupport, I am running these two tests to compare an unset variable:

```liquid
{% if myundefined == empty %}TRUTHY{% else %}NOT TRUTHY{% endif %}
  --> NOT TRUTHY
```

```liquid
{% if myundefined == blank %}TRUTHY{% else %}NOT TRUTHY{% endif %}
  --> TRUTHY
```

**Comment by fw42** [comment]
yeah, undefined (i.e. a Ruby `nil`) is blank, according to ActiveSupport's definition of blank. See [here](http://guides.rubyonrails.org/active_support_core_extensions.html#blank-questionmark-and-present-questionmark).

So no I don't think this is a bug..

**Comment by mikebridge** [comment]
IMHO the [Rails implementation of `empty`](http://stackoverflow.com/questions/885414/a-concise-explanation-of-nil-v-empty-v-blank-in-ruby-on-rails#answer-888877) violates the principle of least surprise (and probably Liskov too).  I found that explaining comparison rules to template designers like "why nil is really not empty" was difficult, so I deviated from Shopfiy liquid in my C# port when it came to comparing values using nil/empty/present/blank.

In fact, I'd love to see the `== [blank|empty|present]` deprecated in favour of `.[blank|empty|present]?` because the comparison isn't really testing equality.  I also thought about maybe supporting a predicate like `x is empty`.
